src/support/import-supervisor.js

```diff
@@ -225,12 +242,20 @@ function ImportSupervisor(services, config) {
       throw new ErrorWithStatusCode('Job ID is required', 400);
     }
 
-    const job = await dataAccess.getImportJobByID(jobId);
+    let job = await dataAccess.getImportJobByID(jobId);
+    // TODO: Modify importApiKey to hashedApiKey
     // Job must exist, and the import API key must match the one provided
     if (!job || job.getApiKey() !== importApiKey) {
       throw new ErrorWithStatusCode('Not found', 404);
     }
 
+    const metadata = await getApiKeyMetadata(importApiKey);
+    if (!metadata) {
+      throw new ErrorWithStatusCode('Server error', 500);
+    }
+
+    job = { ...job, metadata };
+
     return job;
```

Comment:
For simplicity, suggesting:

```suggestion
    const apiKeyMetadata = await getApiKeyMetadata(importApiKey);

    return { ...job, metadata: apiKeyMetadata || {} };
```

Reply:
So, we do not want metadata to be part of the ImportJobDto?